### PR TITLE
[formrecognizer] fixing error on ModelOperation to include details

### PR DESCRIPTION
**Pull request by kristapratico**
Introduces new models to expose the error shape

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py [hunk 1]
```diff
@@ -3898,3 +3889,150 @@ def from_dict(cls, data):
...
+            target=data.get("target", None),
+            details=[DocumentAnalysisError.from_dict(e) for e in data.get("details")]  # type: ignore
+            if data.get("details") else [],
+            innererror=DocumentAnalysisInnerError.from_dict(data.get("innererror"))  # type: ignore
+            if data.get("innererror") else None
+        )
+
+
+class DocumentAnalysisInnerError(object):
+    """Inner error details.
```
Comment (catalinaperalta) [inline sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:3976]: Nit:
```suggestion
    """Inner error details for the DocumentAnalysisError.
```

File: sdk/formrecognizer/azure-ai-formrecognizer/tests/test_mgmt.py [hunk 2]
```diff
@@ -167,6 +167,7 @@ def test_get_list_operations(self, client):
             error = op.error
             assert error.code
             assert error.message
+            assert error.details
```
Comment (catalinaperalta) [inline sdk/formrecognizer/azure-ai-formrecognizer/tests/test_mgmt.py:170]: Not asserting for the inner error?

**Review by catalinaperalta — APPROVED** [review]
Minor details and a question.

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py [hunk 3]
```diff
@@ -3898,3 +3889,150 @@ def from_dict(cls, data):
...
+
+    :ivar code: Error code.
+    :vartype code: str
+    :ivar message: Error message.
+    :vartype message: str
+    :ivar target: Target of the error.
+    :vartype target: str
+    :ivar details: List of detailed errors.
+    :vartype details: list[~azure.ai.formrecognizer.DocumentAnalysisError]
+    :ivar innererror: Detailed error.
```
[the remark this answers is not in the document]
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:3905]: This follows the service shape. They used to include more detailed information in innererror but have switched to details in v3 it seems. Regardless, this follows the ODataV4 [spec ](http://docs.oasis-open.org/odata/odata-json-format/v4.0/os/odata-json-format-v4.0-os.html#_Toc372793091) so if they decide to include more information in other places we will correctly expose it.

File: sdk/formrecognizer/azure-ai-formrecognizer/tests/test_mgmt.py (hunk 2 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/tests/test_mgmt.py:170]: I've only seen info in error details so far so trying to not create a flaky test here

File: sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_models.py:3905]: Oh interesting, ok well sounds good then.

**Pull request merged**